src/components/ActionsMenuLayout/ActionsMenuLayout.tsx

```diff
@@ -16,30 +20,98 @@ export class ActionsMenuLayout extends React.Component<ActionsMenuLayoutProps> {
   static Divider = ActionsMenuLayoutDivider;
 
   static displayName = 'ActionsMenuLayout';
+  state = {
+    focusedIdx: 0,
+  };
+
+  liRefs = [];
+
+  componentDidMount() {
+    const { focusedIndex } = this.props;
+
+    if (focusedIndex) {
+      this.focusItem(focusedIndex);
+    } else {
+      this.focusItem(this.state.focusedIdx);
+    }
+  }
+
+  focusItem(idx: number) {
+    this.setState(
+      {
+        focusedIdx: idx,
+      },
+      () => {
+        this.liRefs[idx].liRef?.current.focus();
+      },
+    );
+  }
 
   getDataAttributes(mobile: boolean) {
     return {
       [ACTIONS_MENU_DATA_KEYS.mobile]: mobile,
     };
   }
 
+  getNextItemIdx(direction: 1 | -1): number {
+    const nextItem = this.state.focusedIdx + direction;
+
+    if (direction === 1) {
+      return nextItem === this.liRefs.length ? 0 : nextItem;
+    }
+
+    return nextItem < 0 ? this.liRefs.length - 1 : nextItem;
+  }
+
+  _onKeyDown = (e: React.KeyboardEvent<HTMLUListElement>) => {
+    if (e.key === KEYS.ArrowDown || e.key === KEYS.ArrowDownIE) {
+      e.preventDefault(); // remove scroll on space
+      this.focusItem(this.getNextItemIdx(1));
+    }
+
+    if (e.key === KEYS.ArrowUp || e.key === KEYS.ArrowUpIE) {
+      e.preventDefault(); // remove scroll on space
+      this.focusItem(this.getNextItemIdx(-1));
+    }
+  };
+
+  addRef(ref, index: number) {
+    if (ref) {
+      if (
+        this.liRefs.find(_ref => {
+          return _ref?.props.content === ref.props.content;
+        })
+      ) {
+        return;
+      }
+
+      this.liRefs[
+        index === 0 || this.liRefs[index - 1] ? index : index - 1
+      ] = ref;
+    }
+  }
+
   render() {
     const { alignment, children, className } = this.props;
     return (
       <TPAComponentsConsumer>
         {({ mobile }) => (
           <ul
+            onKeyDown={this._onKeyDown}
             className={st(classes.root, { mobile }, className)}
             {...this.getDataAttributes(mobile)}
             data-hook={this.props['data-hook']}
             role="menu"
-            tabIndex={-1}
```

Comment:
I think we should leave the `tabindex="-1"`
and also add an onfocus handler to the `ul`,
which will be something like this:
```js
_onFocus: () => {
    const { focusedIdx } = this.state;
    const { focusedIndex } = this.props;

    this.focusItem(typeof focusedIndex !== 'undefined' ? focusedIndex : focusedIdx);
}
```

Reply:
Yeah, nice catch. Will add!